modules/porous_flow/src/dirackernels/PorousFlowPointEnthalpySourceFromPostprocessor.C

```diff
@@ -0,0 +1,56 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowPointEnthalpySourceFromPostprocessor.h"
+#include "SinglePhaseFluidProperties.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowPointEnthalpySourceFromPostprocessor);
+
+InputParameters
+PorousFlowPointEnthalpySourceFromPostprocessor::validParams()
+{
+  InputParameters params = DiracKernel::validParams();
+  params.addRequiredParam<PostprocessorName>("mass_flux",
+                                             "The postprocessor name holding the mass flux at this "
+                                             "point in kg/s (only positive values are allowed)");
+  params.addRequiredParam<UserObjectName>("fp", "The name of the user object for fluid properties");
+  params.addRequiredCoupledVar("pressure", "Pressure");
```

Comment:
I've written heaps of comments, but this is the main one.... Is "pressure" supposed to be (1) the pressure of the fluid in the porous material, or (2) the pressure of the injected fluid. 

(1) You should get pressure from the PorousFlow material system, and you should provide derivatives of the residual in a computeQpJacobian.

(2) You should state in the documentation (param and markdown) that this is usually an AuxVariable representing the pressure of an external agent, such as a borehole, but that if it is a PorousFlow variable then no Jacobian entries are computed so this DiracKernel may result in slightly slower numerical convergence.

Reply:
Are you saying that the pressure value I am using is not the right one? Because I do use the pressure we solve for, but seems like it should be the pressure of the injected material. If that is so, then does that need to be a user input instead?

Reply:
Upon a bit more reflection, i think you are potentially doing (1) and (2).   So i think the documentation (in param and markdown) should state this.  Something like:

"Pressure (along with T_in) is used to calculate the injected fluid enthalpy.  Most frequently, it is the PorousFlow pressure variable (the porepressure in the porous medium).  This models the situation where fluid is injected at a specified rate and temperature (using this DiracKernel and a PorousFlowPointSourceFromPostprocessor) which potentially leads to changes in the porepressure.  Alternately, it may be the pressure of the injected fluid, as fixed by an external agent (such as a pump) which is stored in an AuxVariable.